uxarray/grid/grid.py

```diff
@@ -96,211 +93,77 @@ def __init__(self, input_obj, **kwargs):
         # initialize cached data structures (nearest neighbor operations)
         self._ball_tree = None
 
-        # unpack kwargs with default values set to None
-        kwargs_list = [
-            'gridspec', 'vertices', 'islatlon', 'isconcave', 'source_grid',
-            'use_dual'
-        ]
-        for key in kwargs_list:
-            setattr(self, key, kwargs.get(key, None))
-
-        # check if initializing from verts:
-        if isinstance(input_obj, (list, tuple, np.ndarray)):
-            input_obj = np.asarray(input_obj)
-            self.mesh_type = "From vertices"
-            # grid with multiple faces
-            if input_obj.ndim == 3:
-                self.__from_vert__(input_obj)
-                self.source_grid = "From vertices"
-            # grid with a single face
-            elif input_obj.ndim == 2:
-                input_obj = np.array([input_obj])
-                self.__from_vert__(input_obj)
-                self.source_grid = "From vertices"
-            else:
-                raise RuntimeError(
-                    f"Invalid Input Dimension: {input_obj.ndim}. Expected dimension should be "
-                    f"3: [nMesh2_face, nMesh2_node, Two/Three] or 2 when only "
-                    f"one face is passed in.")
-
-        # check if initializing from string
-        # TODO: re-add gridspec initialization when implemented
-        elif isinstance(input_obj, xr.Dataset):
-            self.mesh_type = _parse_grid_type(input_obj)
-            self.__from_ds__(dataset=input_obj)
-        else:
-            raise RuntimeError("Dataset is not a valid input type.")
+    @classmethod
```

Comment:
Could you explain to me what making this an @classmethod achieves?

Reply:
Sure! The `@classmethod` decorator allows us to have a function that bounded to a Class, not a Object. This allows to make a `Grid.from_dataset` method that can return an initialized version of a `Grid`. It also means that we can call `Grid.from_dataset` before initializing a `Grid`.

Reply:
I see, gotcha that is useful. Thanks!